### ISSUE TITLE
Rename Number of CPUs in docs to vCPUs

### ISSUE THREAD

**Issue by ksquaredkey** [aws/karpenter#2539]
In the "Supported Labels", there is a line:

karpenter.k8s.aws/instance-cpu | 32 | [AWS Specific] Number of CPUs on the instance

It is unclear whether the "instance-cpu" is Hardware CPUs or vCPUs on HT enabled instances.  vCPUs is the more useful value and looking at the source in "instancetype.go", it does appear to be vCPUs.  Making this explicitly say "Number of vCPUs on the instance" would be clearer.

**Comment by dewjam** [comment]
Hey @ksquaredkey ,
I'm interpreting "CPU" as a generic term to convey the default number of CPU cores visible to the OS (and then reported by Kubelet).  By using "CPU", it covers both virtual and bare metal instance types.  Although, the ec2 API seems to provide a vCPU count for bare metal instances too.

Is this something that is impacting your use of Karpenter?  Or just looking to make docs more accurate?

**Comment by ksquaredkey** [comment]
Just trying to make the docs more accurately reflect what AWS is calling at the API level of the the DescribeInstanceTypes in the vCpuInfo.  Specifically, I do NOT want Cores and by explicitly calling it vCPU it helps avoid the conflation of "CPU" with "Core" that happens, especially with folks used to non-HT processors (I know a lot of old CompSci farts because I am an old CompSci fart...)

In my particular case, I want to a minimum of "*.4xlarge" and my containers work on both Graviton2 and x86_64.  We can't use the "Gt" operator on "karpenter.k8s.aws/instance-size" and since I want to make sure I get the correct minimum vCPUs using "karpenter.k8s.aws/instance-cpu", I went to the source to confirm that "instance-cpu" was using the vCPU and not Core from the vCpuinfo that I am used to see in DescribeInstanceTypes.  By explicitly calling out vCPU in the doc, it eliminates the risk of the "assume" and saves other folks going "Hmm..." and digging in the source.

I'll admit, minor nit and debatable.  It's just that I went through the above exercise and that "Create documentation issue" link was right there conveniently on the page, so I thought I might save other folks the same trip.
